prow/pubsub/reporter/reporter.go

```diff
@@ -121,12 +123,22 @@ func (c *Client) Report(pj *prowapi.ProwJob) ([]*prowapi.ProwJob, error) {
 
 func (c *Client) generateMessageFromPJ(pj *prowapi.ProwJob) *ReportMessage {
 	pubSubMap := findLabels(pj, PubSubProjectLabel, PubSubTopicLabel, PubSubRunIDLabel)
+
+	var number int
+	if (pj.Spec.Refs == nil) || len(pj.Spec.Refs.Pulls) == 0 {
+		number = 0
+	} else {
+		number = pj.Spec.Refs.Pulls[0].Number
+	}
```

Comment:
```
if pj.Spec.Refs != nil) && len(pj.Spec.Refs.Pulls) > 0 {
	number = pj.Spec.Refs.Pulls[0].Number
}
```

Reply:
removed this as we are now adding the refs